docs/source-pytorch/common/hooks.rst

```diff
@@ -212,13 +212,13 @@ with the source of each hook indicated:
     │   ├── [LightningModule]
     │   └── [Strategy]
     │
-    ├── on_fit_end()
-    │   ├── [Callbacks]
-    │   ├── [LightningModule]
-    │   └── [Strategy]
-    │
     └── teardown(stage="fit")
-        └── [Callbacks only]
+        ├── [Strategy]
+        ├── on_fit_end()
+        │   ├── [Callbacks]
+        │   └── [LightningModule]
+        ├── [Callbacks]
+        └── [LightningModule]
```

Comment:
```suggestion
        ├── [Strategy]
        ├── on_fit_end()
        │   ├── [Callbacks]
        │   └── [LightningModule]
        ├── [LightningDataModule]
        ├── [Callbacks]
        └── [LightningModule]
```


ref: If we consider this teardown hook being called after `on_fit_end`, 
https://github.com/Lightning-AI/pytorch-lightning/blob/da7f2f9a9ff116b2316d4475a58d310834b376c6/src/lightning/pytorch/trainer/trainer.py#L1029-L1032

then it also seems to include the teardown for the datamodule.
https://github.com/Lightning-AI/pytorch-lightning/blob/da7f2f9a9ff116b2316d4475a58d310834b376c6/src/lightning/pytorch/trainer/call.py#L132-L141